projects/fw103/src/main.c

```diff
@@ -0,0 +1,61 @@
+#include "adc.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+#include "wait.h"
+
+// Pin A6 Reader Function
+void read_pin_a6(const GpioAddress *address, void *context) {
```

Comment:
Any function that is only used in the file it is declared should be declared as `static void` and prefixed with `prv_`, and it's a good idea to put `_cb` or `_callback` at the end so that we know that it will be registered as a callback

This helps us keep track which functions are part of an interface and which are used only by other methods in the same file.